### PR TITLE
More instanceof pattern matching

### PR DESCRIPTION
**Pull request by holgerfriedrich**
Some clean-up.

### CODE REVIEW

File: bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleRegistryImpl.java [hunk 1]
```diff
@@ -589,7 +589,7 @@ private boolean checkType(Type type, Object configValue) {
                 return configValue instanceof Boolean;
             case INTEGER:
                 return configValue instanceof BigDecimal || configValue instanceof Integer
-                        || configValue instanceof Double && ((Double) configValue).intValue() == (Double) configValue;
+                        || configValue instanceof Double doubleValue && doubleValue.intValue() == (Double) configValue;
```
Comment (J-N-K) [inline bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleRegistryImpl.java:592]:
```suggestion
                        || configValue instanceof Double doubleValue && doubleValue.intValue() == doubleValue;
```

**Review by J-N-K — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleRegistryImpl.java (hunk 1 above)
Comment (holgerfriedrich) [reply bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleRegistryImpl.java]: done

**Review by J-N-K — APPROVED** [review]
Thanks!

**Pull request merged**